runtime/src/lib.rs

```diff
@@ -272,15 +273,14 @@ pub trait Listener: Sync + Send + 'static {
 /// messages over a network connection.
 pub trait Sink: Sync + Send + 'static {
     /// Send a message to the sink.
-    fn send(&mut self, msg: &[u8]) -> impl Future<Output = Result<(), Error>> + Send;
+    fn send(&mut self, msg: Bytes) -> impl Future<Output = Result<(), Error>> + Send;
 }
 
 /// Interface that any runtime must implement to receive
 /// messages over a network connection.
 pub trait Stream: Sync + Send + 'static {
-    /// Receive a message from the stream, storing it in the given buffer.
-    /// Reads exactly the number of bytes that fit in the buffer.
-    fn recv(&mut self, buf: &mut [u8]) -> impl Future<Output = Result<(), Error>> + Send;
+    /// Reads exactly `length` bytes from the stream and returns them.
+    fn recv(&mut self, length: usize) -> impl Future<Output = Result<Bytes, Error>> + Send;
```

Comment:
I suppose the alternative here would be to provide a `Bytes` of some len (in case the caller was reusing buffers)?